packages/cli/src/sso.ee/saml/saml.service.ee.ts

```diff
@@ -236,20 +241,58 @@ export class SamlService {
 		};
 	}
 
-	async setSamlPreferences(prefs: Partial<SamlPreferences>): Promise<SamlPreferences | undefined> {
+	async setSamlPreferences(
+		prefs: Partial<SamlPreferences>,
+		tryFallback: boolean = false,
+	): Promise<SamlPreferences | undefined> {
 		await this.loadSamlify();
 		await this.loadPreferencesWithoutValidation(prefs);
 		if (prefs.metadataUrl) {
-			const fetchedMetadata = await this.fetchMetadataFromUrl();
-			if (fetchedMetadata) {
-				this._samlPreferences.metadata = fetchedMetadata;
+			try {
+				const fetchedMetadata = await this.fetchMetadataFromUrl();
+				if (fetchedMetadata) {
+					this._samlPreferences.metadata = fetchedMetadata;
+				} else {
+					// in this case the metadata url didn't produce a valid metadata for SAML
+					// therefore we are rejecting the change to it
+					throw new InvalidSamlMetadataUrlError();
+				}
+			} catch (error) {
+				if (tryFallback) {
+					// we were not able to produce correct metadata from the URL, but
+					// in this case we don't care and try to fallback on the saved metadata in the
+					// database.
+					this.logger.warn(
+						'SAML initialization detected an invalid metadata URL in database. Trying to initialize from metadata in database if available.',
+					);
+				} else {
+					throw error;
```

Comment:
This is not covered by a test as far as I can tell.

![image](https://github.com/user-attachments/assets/4c02f55c-3d5e-4e83-bf8d-8f6a2d248da7)